File: tkpddigital/src/main/java/com/tokopedia/digital/cart/model/CartAdditionalInfo.java
package com.tokopedia.digital.cart.model;

import java.util.List;

/**
 * Created by Nabilla Sabbaha on 3/1/2017.
 */

public class CartAdditionalInfo {

    private String title;

    private List<CartItemDigital> cartItemDigitalList;

    public CartAdditionalInfo(String title, List<CartItemDigital> cartItemDigitalList) {
        this.title = title;
        this.cartItemDigitalList = cartItemDigitalList;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<CartItemDigital> getCartItemDigitalList() {
        return cartItemDigitalList;
    }

    public void setCartItemDigitalList(List<CartItemDigital> cartItemDigitalList) {
        this.cartItemDigitalList = cartItemDigitalList;
    }
}

File: features/recharge/digitalproduct/src/main/java/com/tokopedia/digital/newcart/presentation/model/cart/CartAdditionalInfo.kt
package com.tokopedia.digital.newcart.presentation.model.cart

import android.os.Parcel
import android.os.Parcelable

import com.tokopedia.abstraction.base.view.adapter.Visitable

import java.util.ArrayList

/**
 * @author by Nabilla Sabbaha on 3/1/2017.
 */

class CartAdditionalInfo(
        var title: String? = null,

        var cartItemDigitalList: List<CartItemDigital>? = null
) : Parcelable, Visitable<Void> {

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.createTypedArrayList(CartItemDigital)) {
    }

    override fun type(typeFactory: Void): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeTypedList(cartItemDigitalList)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<CartAdditionalInfo> {
        override fun createFromParcel(parcel: Parcel): CartAdditionalInfo {
            return CartAdditionalInfo(parcel)
        }

        override fun newArray(size: Int): Array<CartAdditionalInfo?> {
            return arrayOfNulls(size)
        }
    }
}
